XcodeServerSDKTests/XcodeServerEndpointsTests.swift

```diff
@@ -24,5 +25,333 @@ class XcodeServerEndpointsTests: XCTestCase {
+        let paramsArray = [
+            [
+                "bot": "botValue"
+            ],
+            [
+                "otherKey": "otherValue",
+                "bot": "botValue"
+            ]
+        ]
+        for params in paramsArray {
+            let url = self.endpoints?.endpointURL(.Bots, params: params)
+            XCTAssertEqual(url!, "/api/bots/botValue", "endpointURL(.Bots, \(params)) should return \"/api/bots/botValue\"")
+        }
+    }
+    
+    func testEndpointURLCreationForBotsBotRev() {
+        let paramsArray = [
+            [
+                "rev": "revValue",
+                "bot": "botValue"
+            ],
+            [
+                "otherKey": "otherValue",
+                "bot": "botValue",
+                "rev": "revValue"
+            ]
+        ]
+        for params in paramsArray {
+            let url = self.endpoints?.endpointURL(.Bots, params: params)
+            XCTAssertEqual(url!, "/api/bots/botValue/revValue", "endpointURL(.Bots, \(params)) should return \"/api/bots/botValue/revValue\"")
+        }
+    }
+    
+    // MARK: endpointURL(.Integrations)
+    
+    func testEndpointURLCreationForIntegrationsWithoutParams() {
+        let url = self.endpoints?.endpointURL(.Integrations)
+        XCTAssertEqual(url!, "/api/integrations", "endpointURL(.Integrations) should return \"/api/integrations\"")
+    }
+    
+    func testEndpointURLCreationForBotsBotIntegrations() {
+        let paramsArray = [
+            [
+                "bot": "botValue"
+            ],
+            [
+                "otherKey": "otherValue",
+                "bot": "botValue"
+            ],
+            [
+                "otherKey": "otherValue",
+                "bot": "botValue",
+                "integration": "integrationValue"
+            ]
+        ]
+        for params in paramsArray {
+            let url = self.endpoints?.endpointURL(.Integrations, params: params)
+            XCTAssertEqual(url!, "/api/bots/botValue/integrations", "endpointURL(.Integrations, \(params)) should return \"/api/bots/botValue/integrations\"")
+        }
+    }
+    
+    func testEndpointURLCreationForBotsBotRevIntegrations() {
+        let paramsArray = [
+            [
+                "rev": "revValue",
+                "bot": "botValue"
+            ],
+            [
+                "otherKey": "otherValue",
+                "bot": "botValue",
+                "rev": "revValue"
+            ],
+            [
+                "otherKey": "otherValue",
+                "bot": "botValue",
+                "rev": "revValue",
+                "integration": "integrationValue"
+            ]
+        ]
+        for params in paramsArray {
+            let url = self.endpoints?.endpointURL(.Integrations, params: params)
+            XCTAssertEqual(url!, "/api/bots/botValue/revValue/integrations", "endpointURL(.Integrations, \(params)) should return \"/api/bots/botValue/revValue/integrations\"")
```

Comment:
Hmm, that looks weird. I would expect that url to be just `/api/bots/botValue/integrations`. AFAIK, `rev` is only used when deleting a bot. Maybe you found a bug :laughing: